Objects/unionobject.c

```diff
@@ -368,6 +387,151 @@ _Py_union_args(PyObject *self)
     return ((unionobject *) self)->args;
 }
 
+static PyObject *
+union_new(PyTypeObject *type, PyObject *args, PyObject *kwds)
+{
+    if (!_PyArg_NoKeywords("UnionType", kwds)) {
+        return NULL;
+    }
+    return _Py_union_from_tuple(args);
+}
+
+static PyObject *
+call_typing_func_object(const char *name, PyObject **args, size_t nargs)
```

Comment:
I guess there's no obvious header in which to expose this to avoid duplicating it...

Reply:
I was thinking of putting typevarobject.c and unionobject.c in a single .c file, but typevarobject.c is 1600 lines already.